crates/nu-cli/tests/completions/mod.rs

```diff
@@ -363,6 +396,115 @@ fn file_completions() {
+    // Create a new engine
+    let (dir, dir_str, mut engine, mut stack) = new_engine();
+    let command = r#"def list [ ...args: any ] {}"#;
+    assert!(support::merge_input(command.as_bytes(), &mut engine, &mut stack).is_ok());
+
+    // Instantiate a new completer
+    let mut completer = NuCompleter::new(Arc::new(engine), Arc::new(stack));
+
+    // Test completions for the current folder
+    let target_dir = format!("list {dir_str}{MAIN_SEPARATOR}");
+    let suggestions = completer.complete(&target_dir, target_dir.len());
+
+    // Create the expected values
+    let expected_paths: Vec<String> = vec![
+        folder(dir.join("another")),
+        file(dir.join("custom_completion.nu")),
+        folder(dir.join("directory_completion")),
+        file(dir.join("nushell")),
+        folder(dir.join("test_a")),
+        folder(dir.join("test_b")),
+        file(dir.join(".hidden_file")),
+        folder(dir.join(".hidden_folder")),
+    ];
+
+    #[cfg(windows)]
+    {
+        let separator = '/';
+        let target_dir = format!("list {dir_str}{separator}");
+        let slash_suggestions = completer.complete(&target_dir, target_dir.len());
+
+        let expected_slash_paths: Vec<String> = expected_paths
+            .iter()
+            .map(|s| s.replace('\\', "/"))
+            .collect();
+
+        match_suggestions(&expected_slash_paths, &slash_suggestions);
+    }
+
+    // Match the results
+    match_suggestions(&expected_paths, &suggestions);
+
+    let target_dir = format!("list somefile.txt {dir_str}{MAIN_SEPARATOR}");
+    let suggestions = completer.complete(&target_dir, target_dir.len());
+
+    // Create the expected values
+    let expected_paths: Vec<String> = vec![
+        folder(dir.join("another")),
+        file(dir.join("custom_completion.nu")),
+        folder(dir.join("directory_completion")),
+        file(dir.join("nushell")),
+        folder(dir.join("test_a")),
+        folder(dir.join("test_b")),
+        file(dir.join(".hidden_file")),
+        folder(dir.join(".hidden_folder")),
+    ];
+
+    #[cfg(windows)]
+    {
+        let separator = '/';
+        let target_dir = format!("list {dir_str}{separator}");
+        let slash_suggestions = completer.complete(&target_dir, target_dir.len());
+
+        let expected_slash_paths: Vec<String> = expected_paths
+            .iter()
+            .map(|s| s.replace('\\', "/"))
+            .collect();
+
+        match_suggestions(&expected_slash_paths, &slash_suggestions);
+    }
+
+    // Match the results
+    match_suggestions(&expected_paths, &suggestions);
+
+    // Test completions for a file
+    let target_dir = format!("list {}", folder(dir.join("another")));
+    let suggestions = completer.complete(&target_dir, target_dir.len());
+
+    // Create the expected values
+    let expected_paths: Vec<String> = vec![file(dir.join("another").join("newfile"))];
+
+    // Match the results
+    match_suggestions(&expected_paths, &suggestions);
```

Comment:
This, too, seems a little unnecessary. There are checks above for:
1. `CUSTOMCOMMAND ARG1 DIR/<TAB>`
2. `CUSTOMCOMMAND DIR/<TAB>`
It seems like those are enough to ensure that file completions are triggered for custom commands. The other test function, `file_completions()`, tests specific things about file completion behavior.

This is not a hill I'm going to die on, though, I'm willing to be convinced otherwise :)

Reply:
Same answer as https://github.com/nushell/nushell/pull/14481#discussion_r1865282933 

Will delete it then.